src/Analyzers/CSharp/CodeFixes/RemoveUnnecessaryLambdaExpression/CSharpRemoveUnnecessaryLambdaExpressionCodeFixProvider.cs

```diff
@@ -52,8 +58,32 @@ protected override Task FixAllAsync(
 
                     return current;
                 });
+
+            // If the inner invocation has important trivia on it, move it to the container of the anonymous function.
+            if (TryGetAnonymousFunctionInvocation(anonymousFunction, out var invocation, out _) &&
+                invocation.GetLeadingTrivia().Any(t => t.IsSingleOrMultiLineComment()))
+            {
+                var containingStatement = anonymousFunction.AncestorsAndSelf().OfType<StatementSyntax>().FirstOrDefault();
+                if (containingStatement != null)
+                {
+                    editor.ReplaceNode(containingStatement,
+                        (current, generator) => current
+                            .WithPrependedLeadingTrivia(TakeComments(invocation.GetLeadingTrivia()))
+                            .WithAdditionalAnnotations(Formatter.Annotation));
+                }
+            }
         }
 
         return Task.CompletedTask;
     }
+
+    private static IEnumerable<SyntaxTrivia> TakeComments(SyntaxTriviaList triviaList)
+    {
+        var lastComment = triviaList.Last(t => t.IsSingleOrMultiLineComment());
+        var lastIndex = triviaList.IndexOf(lastComment) + 1;
+        if (lastIndex < triviaList.Count && triviaList[lastIndex].IsEndOfLine())
+            lastIndex++;
+
+        return triviaList.Take(lastIndex);
```

Comment:
didn't quite get why we need this filtering at a glance - mind adding a comment on it?

Reply:
the idea is that we want to grab up through the last comment and the newline that follows and only move those over if it exists.  we don't want to generally move things like when the user only has blank lines above the inner call.  i will doc :)